packages/trader/src/exchange-adapters/CreateExchangeAdapter.ts

```diff
@@ -0,0 +1,50 @@
+import winston from "winston";
+import Web3 from "web3";
+
+const assert = require("assert");
+const { UniswapTrader } = require("./UniswapTrader");
+
+async function createExchangeAdapter(logger: winston.Logger, web3: Web3, dsProxyManager: any, config: any) {
+  assert(config.type, "Exchange adapter must have a type. EG uniswap for a uniswap dex");
+
+  if (config.type === "uniswap") {
+    const requiredFields = ["tokenAAddress", "tokenBAddress"];
+    if (isMissingField(config, requiredFields, logger)) return null;
+
+    // TODO: refactor these to be pulled from a constants file somewhere.
+    const uniswapRouterAddress = "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D";
+    const uniswapFactoryAddress = "0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f";
```

Comment:
Could we use the uniswap SDK to get these for whichever network we're using?

Reply:
I took a look at this but the SDK only has the `FACTORY_ADDRESS` on main net: https://uniswap.org/docs/v2/SDK/other-exports/

Reply:
I see. I'm kinda surprised they don't export that 🤷